File: Signaux/exo2_1.py
import os
import signal

n = input("Entrez un entier : ")

with open("/tmp/entier.txt", "w") as f:
    f.write(n)

try:
    with open("/tmp/affiche.pid", "r") as f:
        pid_dest = int(f.read())
    os.kill(pid_dest, signal.SIGUSR1)
except FileNotFoundError:
    print("Erreur : Le programme affiche.py n'est pas lancé.")

File: Signaux/exo2_1.c
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>

int main() 
{
    int n, pid_dest;
    printf("Entrez un entier : ");
    scanf("%d", &n);

    // Écriture de la donnée
    FILE *f = fopen("/tmp/entier.txt", "w");
    fprintf(f, "%d", n);
    fclose(f);

    // Lecture du PID du destinataire
    FILE *f_pid = fopen("/tmp/affiche.pid", "r");
    if (f_pid) 
    {
        fscanf(f_pid, "%d", &pid_dest);
        fclose(f_pid);
        kill(pid_dest, SIGUSR1); 
    }
    return 0;
}